ihatemoney/forms.py

```diff
@@ -44,6 +45,25 @@ def process_formdata(self, value):
         return super(CommaDecimalField, self).process_formdata(value)
 
 
+class CalculatorStringField(StringField):
+    """
+    A class to deal with math ops (+, -, *, /)
+    in StringField
+    """
+
+    def process_formdata(self, valuelist):
+        if valuelist:
+            error_msg = "Not a valid amount or expression"
+            value = str(valuelist[0]).replace(",", ".")
+
+            if not match(r'^[ 0-9\.\+\-\*/\(\)]{0,50}$', value) or "**" in value:
```

Comment:
The 50 char limit was more of a guesstimate as to what the limit should be, a pretty low one in hindsight =). We should def enforce a limit to avoid the `**` issue. I think bumping it up to 200 should be more than what most people would need.